datastore/datastore.go

```diff
@@ -0,0 +1,303 @@
+		IntervalEndTime(endTime.Format(time.RFC3339)).
+		Do()
+
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+
+}
+
+func getStackdriverData(resp *monitoring.ListTimeSeriesResponse) ([]map[string]interface{}, error) {
+	var stackdriverMetricBody StackdriverMetric
+	var metricResult []map[string]interface{}
+
+	err := json.Unmarshal(formatResource(resp), &stackdriverMetricBody)
+	if err != nil {
+		return nil, err
+	}
+
+	for id := range stackdriverMetricBody.TimeSeries {
+		metricResult = append(metricResult, map[string]interface{}{
+			"event_type":                        "DatastoreSample",
+			"provider":                          "datastoreStackdriver",
+			"datastoreStackdriver.apiMethod":    stackdriverMetricBody.TimeSeries[id].Metric.Labels.ApiMethod,
+			"datastoreStackdriver.responseCode": stackdriverMetricBody.TimeSeries[id].Metric.Labels.ResponseCode,
+			"datastoreStackdriver.metricType":   stackdriverMetricBody.TimeSeries[id].Metric.Type,
+			"datastoreStackdriver.metricKind":   stackdriverMetricBody.TimeSeries[id].MetricKind,
+			"datastoreStackdriver.timestamp":    stackdriverMetricBody.TimeSeries[id].Points[0].Interval.StartTime.Unix(),
+			"datastoreStackdriver.value":        stackdriverMetricBody.TimeSeries[id].Points[0].Value.Int64Value,
+			"datastoreStackdriver.projectId":    stackdriverMetricBody.TimeSeries[id].Resource.Labels.ProjectID,
+			"datastoreStackdriver.resourceType": stackdriverMetricBody.TimeSeries[id].Resource.Type,
+		})
+	}
+
+	return metricResult, nil
+}
+
+func getDatastoreQueryResult(ds DatastoreImpl) ([]DatastoreKind, error) {
+	q := datastore.NewQuery("__Stat_Kind__").Order("kind_name")
+
+	var kinds []DatastoreKind
+
+	ctx := context.Background()
+	_, err := ds.GetAll(ctx, q, &kinds)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return kinds, nil
+}
+
+func getDatastoreData(kinds []DatastoreKind, projectID string) []map[string]interface{} {
+	var datastoreData []map[string]interface{}
+	for _, k := range kinds {
+		datastoreData = append(datastoreData, map[string]interface{}{
+			"event_type":                         "DatastoreSample",
+			"provider":                           "datastoreQuery",
+			"datastoreQuery.builtinIndexBytes":   k.BuiltinIndexBytes,
+			"datastoreQuery.builtinIndexCount":   k.BuiltinIndexCount,
+			"datastoreQuery.compositeIndexBytes": k.CompositeIndexBytes,
+			"datastoreQuery.compositeIndexCount": k.CompositeIndexCount,
+			"datastoreQuery.entityBytes":         k.EntityBytes,
+			"datastoreQuery.bytes":               k.Bytes,
+			"datastoreQuery.count":               k.Count,
+			"datastoreQuery.kindName":            k.KindName,
+			"datastoreQuery.projectId":           projectID,
+			"datastoreQuery.timestamp":           k.Timestamp.Unix(),
+		})
+	}
+	return datastoreData
+}
+
+func NewClient(projectId string) Client {
+
+	ctx := context.Background()
+	dsClient, err := datastore.NewClient(ctx, projectId)
+	if err != nil {
+		panic("unable to set client")
```

Comment:
These shouldn't be panics, rather return an error at this level. Higher up the stack if appropriate it can be a full panic.